ckanext/geodatagov/search.py

```diff
@@ -0,0 +1,50 @@
+import logging
+from ckan.common import config
+# from ckan.lib.search.common import make_connection
+# from ckan.lib.search.query import SearchQuery
+
+from ckan.lib.search import make_connection, PackageSearchQuery, SolrSettings
+
+
+log = logging.getLogger(__name__)
+
+
+class GeoPackageSearchQuery(PackageSearchQuery):
+    def get_count(self):
+        """
+        Return the count of all indexed packages.
+        """
+        query = "*:*"
+        fq = "+site_id:\"%s\" " % config.get('ckan.site_id')
+        fq += "+state:active "
+
+        conn = make_connection()
+        
+        try:
+            data = conn.search(query, fq=fq, rows=0)    
+        except Exception, e:
+            error = 'Error in GeoPackageSearchQuery.get_count: {}'.format(e)
+            log.error(error)
+            print(error)
+        
+        return data.hits
+
+    def get_paginated_entity_name_modtime(self, max_results=1000, start=0):
+        """
+        Return a list of the name and metadata_modified s of indexed packages.
+        """
+        query = "*:*"
+        fq = "+site_id:\"%s\" " % config.get('ckan.site_id')
+        fq += "+state:active "
+
+        conn = make_connection()
+        try:
+            data = conn.search(query, fq=fq, rows=max_results, fields='name,metadata_modified', start=start)
```

Comment:
Can you add a `sort` (maybe via `metadata_created asc`) so that the pagination returns consistent results? Just in case a new package gets added, having a sort ensures that the new package appears at the end of the sort instead of the middle which might throw off records.

Reply:
Done [here](https://github.com/GSA/ckanext-geodatagov/pull/165/commits/9bde607d42ceb1bccfc17dc24d9c5f4fb81b3c1a#diff-a0207d668e39922cc8203484323f0151R47)